### ISSUE TITLE
Staging Review finding: Screen reader - Safari/Voiceover - form errors not announced

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#114355]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Lifestage Pension and Burial
**Product:** Pension and Burial
**Feature:** Income and Asset Statement 21P-0969

## Findings details
**VA.gov Experience Standard - issue:** [User can perceive all meaningful elements.](https://design.va.gov/about/experience-standards/#user-can-perceive-all-meaningful-elements)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @rsmithadhoc (Accessibility)

### Description

 Throughout the form (text inputs and radios), when skipping a required field, the error message that popped up was not read out. When trying to continue, focus was sent to the input group, but it still did not read out the error message.

This was on Safari with VoiceOver.

<details><summary>Link, screenshot or steps to recreate</summary>

It happened throughout the form, I didn't record all instances of it, but included some in the example video attached.

https://github.com/user-attachments/assets/b8bfdb35-ba53-4888-9b1f-40728293d784

</details>

## Recommended action

Ensure focus is being set to the first invalid field. With the VADS component, the error message should be read out.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** [WCAG 4.1.3 Status Messages - Level AA](https://www.w3.org/TR/WCAG22/#status-messages)
- **Modality:** N/A
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by jerekshoe** [comment]
@shiragoodman This is part of the forms system. Shouldn't the form system team be updating this?

**Comment by rsmithadhoc** [comment]
> [@shiragoodman](https://github.com/shiragoodman) This is part of the forms system. Shouldn't the form system team be updating this?

@jerekshoe We review the product as given to us and log any issues we find. We still expect teams to test their own work and resolve problems themselves. If an issue arises in forms, teams must work with the forms team and file an issue with them, this is not our responsibility. Once it’s determined to be their responsibility and an issue has been created, you can close this ticket.

**Comment by davidakennedy** [comment]
Could we make a fix in the forms library/design system and contribute that back, so this is solved for all teams? Or is that difficult because it's contextual to the type of questions/fields implemented?

**Comment by matt4su** [comment]
Based on @Mottie's feedback added the team-platform-design-system team  label for their awareness.

**Comment by sanjabaj2** [comment]
@humancompanion-usds We didn't get to resolution with Ryan Smith about this ticket but team did internal investigation and firmly believe this is design system issue. We added Platform Design System Team label and we intend to close this and treat it as not launch blocking.

**Comment by humancompanion-usds** [comment]
It's been an ongoing issue in the component. I don't believe we can solve it (we were not able to previously and Safari has not changed their behavior here). It sounds like Robin and DK worked out a work around. That is the likely solution here, which we can add to the Forms System but, it will take some time (I was not aware they had a workaround until today). So you can close this but our users will suffer in the meantime. Are you not able to apply the same logic to your code that Robin did?

**Comment by amyleadem** [comment]
After some testing, it looks like the screen reader announcement of the error message issue is likely tied to two issues:
1. **The error message span must already be present on the page and receive updates to be announced.** This appears to be something we can fix on the design system side. This is because currently the DS form components appear to be adding the error message span only when the error state is activated, rather than just updating the span's content. 
    -  [MDN's docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/alert_role) explain why this would work: 

        > "As with all other live regions, alerts will only be announced when the content of the element with role="alert" is updated. Make sure that the element with the role is present in the page's markup first - this will "prime" the browser and screen reader to keep watching the element for changes. After this, any changes to the content will be announced. Do not try to dynamically add/generate an element with role="alert" that is already populated with the alert message you want announced - this generally does not lead to an announcement, as it is not a content change."
2. **Focus cannot be moved to anything other the error message.** To reliably announce the alert content, the focus must stay on either the button that triggered the error state (Submit/continue button) or move to the error message itself. If it moves to anywhere else at the same time that the error message is updated, the element that receives focus is what will be announced. 

To demonstrate these findings, I created a [test story](https://114355-error-message-readout--65a6e2ed2314f7b8f98609d8.chromatic.com/?path=/story/uswds-va-text-input--test-toggle-error-state-with-focus-management) with a new focus management control. This should show that: 
1. The screen reader will announce content when the error message element is already on the page
2. The screen reader behavior when focus is moved to different locations on submit (the header, the error message, and no movement) 
    - Details about expected outcomes can be found in the description for the [related demo PR](https://github.com/department-of-veterans-affairs/component-library/pull/1720). 

To address the design system side of this, I've opened https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4616. This should fix the behavior of the alert role so that the error announcements will work as expected _when focus management is absent_. 

If you have any questions or comments, please let us know!

**Comment by aplatt-coforma** [comment]
Per the slack thread, https://dsva.slack.com/archives/C01DBGX4P45/p1754413828133049, we (Robin on Lifestage Benefits Crew) will take ownership on the fix for this issue at the component level and 

Given that this issue is a component issue, we will remove the launch blocking label from this ticket. We will proceed with out launch plans and fix this issue within the next few sprints as Robin as capacity. 

cc @sanjabaj2 @matt4su @mjknight50
